src/chainstate/coordinator/mod.rs

```diff
@@ -760,7 +770,7 @@ impl<
                             "Atlas: {} attachment instances emitted from events",
                             attachments_instances.len()
                         );
-                        match self.attachments_tx.send(attachments_instances) {
+                        match self.attachments_tx.try_send(attachments_instances) {
```

Comment:
What will be the consequences of this failing due to a full channel?  I think it would cause the Atlas subsystem to miss attachments, right?  The `attachment_instances` here are gleaned from `name-register`, `name-update`, and `name-renewal` contract-calls, so if these get dropped, then there's no way for the Atlas subsystem to discover them.

Would it be better to just get rid of the channel, and instead just write the unfulfilled attachment instances directly to the Atlas DB?

Reply:
If you don't want to do this in this PR, can you at least open an issue for it?

Reply:
I believe that the consequences of this being dropped are low: the atlas subsystem will reread the attachments from the database anyways. This was discussed in a blockchain meeting before this PR was initially drafted -- I think @lgalabru mentioned that the Atlas subsystem was designed to allow for these events to be dropped. However, we would need to confirm that that's the case by looking at the codebase. @lgalabru, do you think that these events occassionally getting dropped is okay?

Reply:
What happened with this question?

Reply:
Hmmm. So the receiver component of this channel, located in the p2p thread, is in charge of persisting these attachments instances (not resolved at this point). If some attachments instances are not resolved when a stacks-node is being shutdown, then there's a builtin mechanism that will ensure that we resolve them. If the stacks-node is being shut down while these events are still in the channel, then they only exists in memory and if we're looking at this case, I guess the only way to ensure that we resolve these attachments, is to re-process these blocks.
We could move the logic that persist the attachments in this thread, however it would mean that we now have 2 threads writing to the atlas db, which is probably something we'd like to avoid?

Reply:
The attachement channel being filled (current capacity set to 1) would imply that the chain coordinator was able to process 2 different blocks, with a starving p2p thread, right? Is this possible by design?

Reply:
It makes sense. In that case the method `enqueue_new_attachments` in https://github.com/stacks-network/stacks-blockchain/blob/6c5349b0422c5c88db02d578720467c16b102ab2/src/net/p2p.rs#L5271-L5287 would require some refactoring, but could be a good starting point.

Reply:
Hmm. It seems like changing this to do the DB write in the coordinator thread wouldn't be entirely trivial: there's also the issue of notifying the network result of any now-resolved attachments, and it's not obvious to me if we could just write the the new attachment instances and then let the downloader thread figure out that there are new instances that can be fulfilled (I don't think so).

I think perhaps the best way forward is to deal with the Atlas synchrony issues in another PR -- the current release implementation doesn't retain all events in a shutdown, and worse, can prevent shutdown from occurring smoothly; in most cases this results in a process kill and a corrupted chainstate or API state.

The `try_send` approach resolves the second issue, but it _can_ lead to dropped events even outside of shutdown, so I think the best interim solution is to increase the size of the channel to 5 allowing the coordinator to run 5 blocks ahead of the attachments receiver, and switch this back to a synchronous `send`. In the normal execution, there won't be dropped events, and in the event of a shutdown, as long as the attachments receiver isn't already 5 blocks behind, the coordinator will be able to smoothly shutdown (albeit dropping some Atlas events, but as we already covered: Atlas events are already lost during shutdown). Even in the event that the attachments receiver _was_ behind, fixing the p2p thread panic (which this PR did) should also help here: the mpsc sync_channel implementation should give a channel closed error to the `send()` caller.

Does that make sense to you, @jcnelson and @lgalabru ?

Reply:
(Also, this is what I did in 659b2c45c)

Reply:
> I think the best interim solution is to increase the size of the channel to 5 

Sounds good to me

> it's not obvious to me if we could just write the the new attachment instances and then let the downloader thread figure out that there are new instances that can be fulfilled (I don't think so).

There could be a few ways to approach this. The coordinator could be inserting some entries in the table `attachment_instances`, setting the `created_at`, and the atlas state machine in the p2p thread could keep a value of the most recent `attachment_instances` processed, and only insert the latest ones in the state machine.

Reply:
Damn, wish I'd seen this before merging.  Why stop at 5?  Why not make it, I dunno, 1,000?  It's not like they take much memory.